consensus/src/common.rs

```diff
@@ -51,45 +51,19 @@ impl ConsensusConstants {
     }
 }
 
-/// A chain of [ProtocolStateLegacy]
+/// A chain of ProtocolState
 #[derive(Debug, Default, Eq, PartialEq, Clone)]
 // TODO: replace vec element with ExternalTransition
-pub struct ProtocolStateChain(pub Vec<ProtocolStateLegacy>);
+pub struct ProtocolStateChain<T>(pub Vec<T>)
+where
+    T: ProtocolStateHeader;
 
-/// Trait that represents a chain of block data structure
-pub trait Chain<T>
+impl<T> ProtocolStateChain<T>
```

Comment:
Any particular reason for refactoring these methods out of a trait?

Reply:
I don't see why we would impl this trait for other structs so IMO it's a bit over-designed.  With a trait that support generic, it requires lifetime markers that harms readability a bit